src/coreclr/jit/importercalls.cpp

```diff
@@ -3842,6 +3845,50 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Threading_Volatile_Read:
+            {
+                assert((sig->sigInst.methInstCount == 0) || (sig->sigInst.methInstCount == 1));
+                var_types retType = sig->sigInst.methInstCount == 0 ? JITtype2varType(sig->retType) : TYP_REF;
+#if !TARGET_64BIT
+                if ((retType == TYP_LONG) || (retType == TYP_ULONG) || (retType == TYP_DOUBLE))
```

Comment:
No need to check for `TYP_ULONG`, `JITtype2varType` will never return it (here and below).